cpp/src/arrow/compute/exec/window_functions/window_rank.cc

```diff
@@ -0,0 +1,456 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/compute/exec/window_functions/window_rank.h"
+
+namespace arrow {
+namespace compute {
+
+void WindowRank_Global::Eval(RankType rank_type, const BitVectorNavigator& tie_begins,
+                             int64_t batch_begin, int64_t batch_end, int64_t* results) {
+  int64_t num_rows = tie_begins.bit_count();
+
+  if (rank_type == RankType::ROW_NUMBER) {
```

Comment:
Wouldn't a switch statement be more suitable for this group of `if` statements? Likewise in `WindowRank_Framed1D::Eval()`.